src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto createOrder(Long userId, DoOrderDto shippingAddress) {
+        Order newOrder = orderMapper.toModel(shippingAddress);
+        User user = userRepository.findById(userId).orElseThrow(
+                () -> new RegistrationException("Can't find user by id: " + userId));
+        newOrder.setUser(user);
+        newOrder.setOrderDate(LocalDateTime.now());
+        newOrder.setStatus(Order.Status.PENDING);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartById(userId);
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+        BigDecimal total = cartItems.stream()
+                .map(o -> o.getBook().getPrice().multiply(BigDecimal.valueOf(o.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        newOrder.setTotal(total);
+        orderRepository.save(newOrder);
+        Set<OrderItem> orderItems = getOrderItemsSet(cartItems, newOrder);
+        newOrder.setOrderItems(orderItems);
+        return orderMapper.toDto(newOrder);
+    }
+
+    private Set<OrderItem> getOrderItemsSet(Set<CartItem> cartItems, Order newOrder) {
+        return cartItems.stream()
+                .map(o -> saveOrderItem(o, newOrder))
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem saveOrderItem(CartItem cartItem, Order newOrder) {
+        OrderItem orderItem = new OrderItem();
+        orderItem.setOrder(newOrder);
+        orderItem.setQuantity(cartItem.getQuantity());
+        orderItem.setBook(cartItem.getBook());
+        orderItem.setPrice(cartItem.getBook().getPrice());
+        cartItemRepository.delete(cartItem);
+        return orderItemRepository.save(orderItem);
+    }
+
+    @Override
+    public Set<OrderDto> getHistory(Long userId, Pageable pageable) {
+        Set<Order> orders = orderRepository.findAllByUserId(userId, pageable).toSet();
+        return orders.stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public Set<OrderItemDto> getAllOrderItems(Long orderId, Long userId) {
+        if (getMatch(orderId, userId)) {
+            Order order = orderRepository.findById(orderId)
+                    .orElseThrow(() -> new EntityNotFoundException(
+                            "Can't find order by id: " + orderId));
+            Set<OrderItem> orderItems = order.getOrderItems();
+            return orderItems.stream().map(orderItemMapper::toDto).collect(Collectors.toSet());
+        }
+        return null;
```

Comment:
do not return null from method
delete getMatch method
add custom method `Optional<Order> findByIdAndUserId(Long orderId, Long userId);` to repository